politeiad/backend/gitbe/gitbe.go

```diff
@@ -1157,6 +1158,114 @@ func (g *gitBackEnd) newRecord(token []byte, metadata []backend.MetadataStream,
 	return rm, nil
 }
 
+// getVettedTokens gets the tokens of all vetted records by retrieving the
+// names of the folders in the vetted directory.
+func (g *gitBackEnd) getVettedTokens() ([]string, error) {
+	files, err := ioutil.ReadDir(g.vetted)
+	if err != nil {
+		return nil, err
+	}
+
+	vettedTokens := make([]string, 0, len(files))
+	for _, v := range files {
+		id := v.Name()
+		if !util.IsDigest(id) {
+			continue
+		}
+		vettedTokens = append(vettedTokens, id)
+	}
+
+	return vettedTokens, nil
+}
+
+// getUnvettedTokens gets the tokens of all unvetted records by retrieving the
+// names of the git branches in the unvetted directory.
+func (g *gitBackEnd) getUnvettedTokens() ([]string, error) {
+	branches, err := g.gitBranches(g.unvetted)
+	if err != nil {
+		return nil, err
+	}
+
+	unvettedTokens := make([]string, 0, len(branches))
+	for _, id := range branches {
+		if !util.IsDigest(id) {
+			continue
+		}
+		unvettedTokens = append(unvettedTokens, id)
+	}
+
+	return unvettedTokens, nil
+}
+
+// popuplateTokenPrefixCache popuplates the prefix cache on the gitBackEnd
+// object with the prefixes of the tokens of both vetted and unvetted
+// records. This cache is used to ensure that only tokens with unique prefixes
+// are generated, because this allows lookups based on the prefix of a token.
+func (g *gitBackEnd) popuplateTokenPrefixCache() error {
+	vettedTokens, err := g.getVettedTokens()
```

Comment:
getVettedTokens and getUnvettedTokens reads from shared state without holding the lock.